src/ir/module-utils.cpp

```diff
@@ -48,6 +48,7 @@ Function* copyFunction(Function* func,
                        std::optional<std::vector<Index>> fileIndexMap) {
   auto ret = std::make_unique<Function>();
   ret->name = newName.is() ? newName : func->name;
+  ret->hasExplicitName = newName.is() || func->hasExplicitName;
```

Comment:
```suggestion
  ret->hasExplicitName = func->hasExplicitName;
```
I think it's better to just match the copied function. Their names may differ (if copied into the same module) but the property of having an explicit name seems like it should be the same, at least by default.